### DISCUSSION TITLE
Centering text in the middle/wrapping

### DISCUSSION THREAD

**Discussion by xpresas** [Bodmer/TFT_eSPI#3080] [Q&A - General]
1. Simply Im writing a program that fetches direct messagess from discord bot api and I want to display them on the screen. Ive run into multiple problems which most of them I managed to make it work. (Playing gifs, uploading them to littlefs, swapping gifs and etc).
Tbh I havent checked ALL of the examples because some are outdated and theres so much of them I just thought that Im going to write here and I might get help.
The problem Im facing right now is pretty simple.  I pull a message from discords api. That message contains a handful of words. I cant seem to find a way to 

- wrap text
- display it in the middle of the screen
- consistenly stay at that location. (I tried to use https://github.com/Bodmer/TFT_eSPI/issues/558#issuecomment-593147012 this) and after the loop refreshes the text just moves bit by bit.
- A way to display emojis pulled from that text.

2. Arduino v1.
3. Latest
4. Latest esp32 2.0.14
5. ESP32-Wroom-32U (same old esp, just 38pin version)
6. ST7789_DRIVER (320x240 from aliexpress)
7. SPI

```
    tft.fillScreen(TFT_BLACK);
    tft.setTextColor(TFT_WHITE);
    tft.setTextSize(2);
    tft.setTextDatum(CC_DATUM);
   //this is weird, and I bet there is an easier way to manage this
    tft.drawString("Test1",tft.width() / 2,tft.height() / 2 - 15);
    tft.drawString("Test2",tft.width() / 2,tft.height() / 2 + 0);
    tft.drawString("Test3",tft.width() / 2,tft.height() / 2 + 15);
```
Basically if it would be possible to provide an example of how would I go about centering a text string while also wrapping it. I would appreciate it.

**Comment by Bodmer** [comment]
You will need to write code to split strings at an appropriate point and then plot each segment onto the screen. I do not have further example code to do this.

**Reply by xpresas** [reply]
Could you atleast provide a way to center multiple string in the middle of the screen?
For example how would I go about printing three lines of string and keep them all in the middle without doing some weird pixel/counting manipulation?
